File: client/src/modules/readjson/StatPotion.js
/**
 * Created by Team 2 - LongLH - GDF 17 on 12/12/2022.
 */


var StatPotion = StatCard.extend({

    map: String.prototype,
    action: Map.prototype,
    duration: Number.prototype,
    radiusType: Number.prototype,
    statPerLevel: Map.prototype,
    adjust: Map.prototype,
    healDelay: Number.prototype,

    getRadius: function (rank) {
        return JsonConfig.getInstance().getPotionRadius(this.radiusType).getRadius(rank);
    },

    getPotionValue: function (statType, level) {
        return this.statPerLevel[statType][level];
    },

    getAdjustValue: function (key) {
        return this.adjust[key].value;
    },

    getDuration: function () {
        return this.duration / 1000;
    },

});

var PotionAdjust = cc.Class.extend({

    type: String.prototype,
    value: Number.prototype,

});

File: server/src/readjson/StatPotion.java
package readjson;

import java.util.HashMap;

public class StatPotion extends StatCard {
    private String map;
    private HashMap<String, String> action;
    private int duration;
    private int radiusType;
    private HashMap<String, HashMap<Integer, Integer>> statPerLevel;
    private HashMap<String, PotionAdjust> adjust;
    private int healDelay;

    public double getRadius(int rank) {
        return JsonConfig.getInstance().getPotionRadius(this.radiusType).getRadius(rank);
    }

    public int getPotionValue(String statType, int level) {
        return this.statPerLevel.get(statType).get(level);
    }

    public int getAdjustValue(String key) {
        return this.adjust.get(key).value;
    }

    public double getDuration() {
        return this.duration / 1000.0;
    }
}

class PotionAdjust {
    String type;
    int value;
}